### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by AlbertoBrusa**
Things to test:
- [x] CMRA
- [x] adding packages of each type, including source
- [x] building / starting / testing / linting them (only default templates, should be quick)
- [x] import a source pkg into app / esm-view / package and build / test the app / esm-view / package
- [x] create a template and add a new package with that template
- [x] set various configurations for the sample app / esm-view and make sure it works (ie build it in esbuild mode, build it with a different cdn template and look at the code)

### CODE REVIEW

File: packages/modular-scripts/package.json [hunk 1]
```diff
@@ -84,6 +84,7 @@
     "jest-runner-eslint": "1.1.0",
     "jest-transform-stub": "2.0.0",
     "jest-watch-typeahead": "^2.0.0",
+    "jest-environment-jsdom": "^29.3.1",
```
Comment (cristiano-belloni) [inline packages/modular-scripts/package.json:87]: Can you put it in alphabetical order?

File: packages/eslint-config-modular-app/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-modular-app",
-  "version": "3.0.2",
+  "version": "4.0.0",
```
Comment (cristiano-belloni) [inline packages/eslint-config-modular-app/package.json:3]: This needs to go back to 3.0.2 - unfortunately after `changeset version` you have to revert all versions every time (or once at the end)

File: packages/modular-scripts/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "modular-scripts",
-  "version": "3.6.0",
+  "version": "4.0.2",
```
Comment (cristiano-belloni) [inline packages/modular-scripts/package.json:3]: Same

File: packages/create-modular-react-app/src/index.ts [hunk 4]
```diff
@@ -142,7 +142,7 @@ export default async function createModularApp(argv: {
       'prettier',
       'modular-scripts',
       'eslint-config-modular-app',
-      'typescript@>=4.2.1 <4.5.0',
+      'typescript@^4.8.3',
```
Comment (cristiano-belloni) [inline packages/create-modular-react-app/src/index.ts:145]: Remember to add a changelog to bump CMRA as major, since this is breaking.

Comment (AlbertoBrusa) [reply packages/create-modular-react-app/src/index.ts:145]: Yep makes sense, done

File: packages/eslint-config-modular-app/package.json (hunk 2 above)
Comment (AlbertoBrusa) [reply packages/eslint-config-modular-app/package.json]: Cleaned it up, thanks

**Review by ayseenver — APPROVED** [review]

**Review by sgb-io — APPROVED** [review]

**Review by cristiano-belloni — APPROVED** [review]

**Pull request merged**